test/parallel/test-sqlite.js

```diff
@@ -111,3 +112,101 @@ test('math functions are enabled', (t) => {
     { __proto__: null, pi: 3.141592653589793 },
   );
 });
+
+test('Buffer is supported as the database path', (t) => {
+  const db = new DatabaseSync(Buffer.from(nextDb()));
+  t.after(() => { db.close(); });
+  db.exec(`
+    CREATE TABLE data(key INTEGER PRIMARY KEY);
+    INSERT INTO data (key) VALUES (1);
+  `);
+
+  t.assert.deepStrictEqual(
+    db.prepare('SELECT * FROM data').all(),
+    [{ __proto__: null, key: 1 }]
+  );
+});
+
+test('URL is supported as the database path', (t) => {
+  const url = pathToFileURL(nextDb());
+  const db = new DatabaseSync(url);
+  t.after(() => { db.close(); });
+  db.exec(`
+    CREATE TABLE data(key INTEGER PRIMARY KEY);
+    INSERT INTO data (key) VALUES (1);
+  `);
+
+  t.assert.deepStrictEqual(
+    db.prepare('SELECT * FROM data').all(),
+    [{ __proto__: null, key: 1 }]
+  );
+});
+
+
+suite('URI query params', () => {
```

Comment:
We should also check if a string or Buffer based `file://` URI works.

Reply:
It has landed already. Do you want to add it (in a new PR?).